frontend/src/views/SecretMainPage/components/ActionBar/ActionBar.tsx

```diff
@@ -186,9 +196,88 @@ export const ActionBar = ({
     }
   };
 
+  const onMoveSecrets = async (folderId: string, secretsToMove: { _id: string }[]) => {
+    try {
+      await moveSecretsToFolder({
+        secrets: secretsToMove,
+        folderId,
+        workspaceId,
+        environment,
+        secretPath
+      });
+      resetSelectedSecret();
+      handlePopUpClose("moveSecrets")
+      createNotification({
+        text: `Successfully moved  ${Object.keys(selectedSecrets).length > 1 ? "secrets" : "secret"}`,
+        type: "success"
+      });
+    } catch (error) {
+      createNotification({
+        text: `Failed to move ${Object.keys(selectedSecrets).length > 1 ? "secrets" : "secret"}`,
+        type: "error"
+      });
+    }
+  }
+
+  const handleMoveSecretsModalOpen = useCallback(() => handlePopUpOpen("moveSecrets"), []);
+
+  const checkedSecrets = Object.keys(selectedSecrets).map((secretId) => ({
+    _id: secretId,
+  }));
+
+
   return (
     <>
       <div className="flex items-center space-x-2 mt-4">
+        <div className={twMerge("fixed transform  flex justify-center opacity-0 bottom-[20px] left-[220px] scale-50 right-0 z-10 pointer-events-none translate-y-20  transition-all duration-300", Object.keys(selectedSecrets).length > 0 && "translate-y-0 scale-100 opacity-100")}>
```

Comment:
multiple comments over this one

1. Missing permission check 
2. Avoid defining custom tailwind classes which already exist 
3. Move back to old multi select style please